ggml.c

```diff
@@ -5579,20 +5579,36 @@ struct ggml_tensor * ggml_pool_2d(
         is_node = true;
     }
 
+    struct ggml_tensor * result;
+#if defined(GGML_USE_CUBLAS)
```

Comment:
I don't think it is ok to generate a different graph in `ggml.c` based on whether CUDA is enabled. I understand that the reason is because CUDA doesn't support the `GGML_OP_POOL_2D` operation, but this should be handled in a different way. The same fallback could be applied to Metal, which also doesn't implement `GGML_OP_POOL_2D`.

Reply:
So how should we deal with this part?  ggml_backend_cuda_graph_compute which used in a cuda init backend would assert false when compute Pool_2d node.

Reply:
What's preventing from implementing the operator in CUDA?

Reply:
We don't really have a good way to handle this automatically from the ggml side. For now, I think we could have different functions for the implementation with `GGML_OP_POOL_2D` and for the implementation with im2col. Then in the application, we could choose which function to use depending on the backend being used.

Reply:
Implement CUDA POOL2D in ba5592c65336159e7de379f1e39a1354ce287b74

Reply:
Can you add a test case to `test-backend-ops.cpp` to test this op?

Reply:
Add test case in 1a82788028337d114e7949ca3262ccb3869a5dd6